### PR TITLE
add support for v2 gemm2 a8w8

### PR DESCRIPTION
**Pull request by Bernard-Liu**
## Motivation                                                                                        
                                                                                                       
  Enable per-1x32 MXFP8 A8W8 support in FlyDSL v2 GEMM2, allowing `flydsl_moe2_layout_afp8_wfp8_*`     
  kernels to participate in normal MoE tuning, runtime dispatch, and AOT compilation.                  
                                                                                                       
  ## Technical Details                                                                                 
                                                                                                       
  - Add FP8 B layout/loading and scaled FP8×FP8 MFMA support via `b_dtype`.                            
  - Propagate the B dtype through dispatcher/cache, fused MoE runtime, tuner, and AOT.                 
  - Use runtime-aligned FP8 payloads and E8M0 scales.                                                  
  - Preserve existing A4W4/A8W4 behavior and restrict GEMM2 to supported `BN={128,256}`.               
                                                   
  ## Test Plan                                                                                         
                                                                                                       
  - Run focused layout, scale-coupling, padding, atomic/reduce, NT, and standalone BM16 tests.         
  - Run A4W4/A8W4 IR/ISA regression checks.                                                            
  - Tune all Minimax M3 MXFP8 shapes and validate the resulting configurations through the MoE runtime 
  and AOT path.                                    
                                                                                                       
  ## Test Result                                                                                       
                                                                                                       
  - 14 focused tests passed.                                                                           
  - All 32 tuning shapes completed successfully; v2 candidates passed correctness checks and competed  
  normally with existing kernels.                                                                      
  - A4W4/A8W4 regression, end-to-end correctness, and AOT validation passed.

### CODE REVIEW

File: op_tests/flydsl_tests/mxmoe_v2_test_utils.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
+# SPDX-License-Identifier: MIT
```
Comment (coderfeli) [inline op_tests/flydsl_tests/mxmoe_v2_test_utils.py]: why such strange util? don't put in op_tests.

File: op_tests/flydsl_tests/test_mxmoe_gemm2_mxfp8.py [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+def test_v2_gemm2_compile_rejects_bn64():
+    from aiter.ops.flydsl.kernels.mxmoe_dispatcher import (
+        compile_gemm2_a4w4_port,
+    )
+
+    with pytest.raises(AssertionError, match=r"BN in \{128,256,512\}.*BN=64"):
+        compile_gemm2_a4w4_port(BN=64)
+
+
+def test_v2_gemm2_wrapper_rejects_bn_outside_catalog(monkeypatch):
```
Comment (coderfeli) [inline op_tests/flydsl_tests/test_mxmoe_gemm2_mxfp8.py]: ？？？

File: op_tests/flydsl_tests/test_mxmoe_gemm2_mxfp8.py [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+from aiter.utility.fp4_utils import moe_mxfp4_sort
+from op_tests.flydsl_tests.mxmoe_v2_test_utils import (
+    logits_diff,
+    run_standalone_v2_a8w8,
+)
+
+STANDALONE_LOGITS_DIFF_LIMIT = 1e-5
+
+
+def test_v2_gemm2_compile_rejects_bn64():
```
Comment (coderfeli) [inline op_tests/flydsl_tests/test_mxmoe_gemm2_mxfp8.py]: ？？？

File: op_tests/flydsl_tests/mxmoe_v2_test_utils.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    @property
+    def a_scale(self):
+        return self.scales["a"]
+
+    @property
+    def b_scale(self):
+        return self.scales["b"]
+
+
+def logits_diff(x: torch.Tensor, y: torch.Tensor) -> float:
```
Comment (coderfeli) [inline op_tests/flydsl_tests/mxmoe_v2_test_utils.py]: duplicated?

**Review by coderfeli — CHANGES_REQUESTED** [review]
Clean tests.

**Review by coderfeli — APPROVED** [review]
